apps/meteor/client/views/admin/users/AdminUserFormWithData.tsx

```diff
@@ -16,9 +18,19 @@ type AdminUserFormWithDataProps = {
 	roleError: unknown;
 };
 
+const filterUnavailableRoles = (availableRoles: IRole[] | undefined) => (data: UserInfoQueryData) => {
+	if (!availableRoles) {
+		return data;
+	}
+	return {
+		...data,
+		user: { ...data.user, roles: data.user.roles.filter((role: string) => isTruthy(availableRoles.find(({ _id }) => _id === role))) },
```

Comment:
Considering that `availableRoles` is an array of objects, couldn't we use `availableRoles.some(...)` instead of `isTruthy`?

Reply:
Also, I don't know if this is a concern, but if `availableRoles` array is big enough I'd even consider creating a `Set` before. E.g `const available = new Set(availableRoles.map(({ _id }) => _id)` then using `available.has(role)`.

Reply:
Yep that does make sense. I don't think there would be a difference between find and some performance wise, but it's way easier to read, thanks for catching this one.

I'm not sure it's necessary to make a set. I don't think roles ever exceed 50, and I think up to 100 this wouldn't be an issue. I'll do some manual perf testing just to make sure and come back to you, but I think that won't be necessary

Reply:
@aleksandernsilva too many data transformations between array and `Set` to make it worthy, I think.